src/Microsoft.AspNetCore.Session/SessionMiddleware.cs

```diff
@@ -35,12 +34,14 @@ public class SessionMiddleware
         /// <param name="dataProtectionProvider">The <see cref="IDataProtectionProvider"/> used to protect and verify the cookie.</param>
         /// <param name="sessionStore">The <see cref="ISessionStore"/> representing the session store.</param>
         /// <param name="options">The session configuration options.</param>
+        /// <param name="sessionKeyGenerator">The <see cref="ISessionKeyGenerator"/> representing the session key generator</param>
         public SessionMiddleware(
             RequestDelegate next,
             ILoggerFactory loggerFactory,
             IDataProtectionProvider dataProtectionProvider,
             ISessionStore sessionStore,
-            IOptions<SessionOptions> options)
+            IOptions<SessionOptions> options,
+            ISessionKeyGenerator sessionKeyGenerator)
```

Comment:
Can you please elaborate more on the API breaking. Looking through middleware instantiation sources in the `HttpAbstractions` and `ActivatorUtilities` the only convention for a middleware constructor I can see is for it to have `RequestDelegate` as first parameter.

Considering usage of `UseMiddleware` in `SessionMiddlewareExtensions.cs` it is called without parameters or with single parameter - IOptions<SessionOptions>. Therefore second constructor can possibly cause ambiguity as there will be two constructors with identical matching metrics and the result will depend on the order of them beeing processed. That can cause using constructor with default value while populating through DI will be expected. 

Finally injecting of ISessionKeyGenerator is identical to injection of ISessionStore defined in this package.